src/Core/Services/SendService.cs

```diff
@@ -185,32 +186,36 @@ public async Task<Send> GetAsync(string id)
         public async Task ReplaceAsync(Dictionary<string, SendData> sends)
         {
             var userId = await _userService.GetUserIdAsync();
-            await _storageService.SaveAsync(SEND_KEY(userId), sends);
+            await _storageService.SaveAsync(GetSendKey(userId), sends);
             _decryptedSendsCache = null;
         }
 
         public async Task SaveWithServerAsync(Send send, byte[] encryptedFileData)
         {
 
             var request = new SendRequest(send);
-            SendResponse response = null;
+            SendResponse response;
             if (send.Id == null)
             {
-                if (send.Type == SendType.Text)
-                {
-                    response = await _apiService.PostSendAsync(request);
-                }
-                else if (send.Type == SendType.File)
+                switch (send.Type)
                 {
-                    var fd = new MultipartFormDataContent($"--BWMobileFormBoundary{DateTime.UtcNow.Ticks}")
-                    {
-                        { new StringContent(JsonConvert.SerializeObject(request)), "model" },
-                        { new ByteArrayContent(encryptedFileData), "data", send.File.FileName.EncryptedString }
-                    };
-
-                    response = await _apiService.PostSendFileAsync(fd);
+                    case SendType.Text:
+                        response = await _apiService.PostSendAsync(request);
+                        break;
+                    case SendType.File:
+                        var fd = new MultipartFormDataContent($"--BWMobileFormBoundary{DateTime.UtcNow.Ticks}")
+                        {
+                            { new StringContent(JsonConvert.SerializeObject(request)), "model" },
+                            { new ByteArrayContent(encryptedFileData), "data", send.File.FileName.EncryptedString }
+                        };
+
+                        response = await _apiService.PostSendFileAsync(fd);
+                        break;
+                    default:
+                        response = new SendResponse();
```

Comment:
This default behavior idea might just be a bad one. It's identifying a possible future bug and recovering from it in a hard-to-track-down manner.

I think throwing would be better. Thoughts?

Reply:
yes, please add `throw` here

Reply:
I ended up adding the throw only here. 

I don't think we want to throw whenever we encounter one since that would do bad things like lock up Sync if clients get out of date relative to servers. 

Instead, we should just keep the client from ruining saved data by complaining last minute that it doesn't know what it's doing.